### ISSUE TITLE
Test new drawer toggle pattern for a11y

### ISSUE THREAD

**Issue by kimisgold** [omeka/plugin-Contribution#76]
We're revising Omeka Classic's collapsible drawer pattern and applying it to the plugins. This pattern includes components that can collapse and expand, and often be marked for removal or be dragged to create order. This pattern appears within the Contribution Types form. You can go to Contribution > Contribution Types > Add a Type to access this form, or edit one of the existing types from the Contribution Types view. 

Test the following on screen readers for mobile and desktop browsers:

* Each of the elements can be marked for deletion, and then unmarked.
* Screen readers are alerted when new elements are added.
* Form elements are properly labeled when navigating via keyboard.

The drawers on this form are always open, and we are not looking at sorting functionality. You need to be on the `drawer-toggle-a11y-rebase` branch for Omeka Classic, and the `drawer-toggle-a11y` branch of Contribution.

**Comment by aviannamiller** [comment]
Tested on VoiceOver on MacOS and iOS, TalkBack on Android, and NVDA and Jaws on Windows.

Across screen readers and desktop and mobile, I am having success with marking and unmarking elements for deletion and screen reader alerts when new elements are added.

Form elements generally have proper labelling. However, the remove/undo remove element buttons have a repetitive readout on TalkBack, assumedly from their having a duplicate aria-label and title. Also, the readout for the undo function is oddly coupled with an announcement of either "collapsed" or "expanded" (even though undo is only accessible in a collapsed element state), seemingly dependent on the steps taking before undoing deletion:
- "collapsed" is read if the element had been in an expanded state before being marked for deletion
- "expanded" is read if the element is marked for deletion after collapsing, expanding, and then marking for deletion once more (or so on...)

Of a slightly similar note, the element type dropdown announces its collapsed state but does not, for whatever reason, seem to repeat this action for when it is expanded.

Perhaps a bit outside of the specific criteria you are testing in this moment, but I did want to note the following observations:
- Using the "enter" key to navigate the form via keyboard leads to a bit of a bug when one is in the "Prompt" text field or "Multiple rows" checkbox in which a new element is added below with each key press
- After the "Add Element" button is selected, focus moves to "Save Changes" instead of to the newly added element form.

**Comment by kimisgold** [comment]
Similar to UserProfiles, there were some mixups with `aria-expanded` applications and a missing `type="button"` attribute on the "Add Element" button that was having the form submit unintentionally. The changes are in and are ready for a look.

I'm also looking into managing the focus when elements are added, inline with our discussion in omeka/plugin-UserProfiles#24

**Comment by aviannamiller** [comment]
The newly redirected focus works great / behaves as expected in the workflow here.

I am still having issues with page reloads upon using the enter key as an alternative to control+option+space to make selections on the form.

- When doing so on any available fields of a newly added element, the page reloads with the following error message:
  > Element: You must select an element to contribute.
  > Prompt: You must provide a prompt for your contributors.

- When doing so on any available fields of an existing element, the page reloads and redirects back to the Contribution Types browse page.

**Comment by aviannamiller** [comment]
You referenced the usage of `aria-expanded`, though I should note I am still having the same collapsed/expanded readout issues for both the undo button and the element type dropdown menu. Was a fix pushed for these specific problems that I should be seeing changes from?

**Comment by aviannamiller** [comment]
Great, the required attributes look good / have proper screen reader announcements.

**Comment by kimisgold** [comment]
Awesome. Is there anything left on this issue before it can be closed?
